app/models/amusement_park.rb

```diff
@@ -1,3 +1,10 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+
+  def mech_name
+    self.rides
+        .joins(:mechanics)
+        .distinct
+        .pluck('mechanics.name')
+  end
```

Comment:
This query is great! Love the `joins` here. How would you be able to solve this with another association instead of a `joins`?